src/data/set/basic.lean

```diff
@@ -1323,6 +1323,35 @@ theorem prod_mono {s₁ s₂ : set α} {t₁ t₂ : set β} (hs : s₁ ⊆ s₂)
   set.prod s₁ t₁ ⊆ set.prod s₂ t₂ :=
 assume x ⟨h₁, h₂⟩, ⟨hs h₁, ht h₂⟩
 
+lemma prod_subset_prod_iff :
+  (set.prod s t ⊆ set.prod s₁ t₁) ↔ (s ⊆ s₁ ∧ t ⊆ t₁) ∨ (s = ∅) ∨ (t = ∅) :=
+begin
+  classical,
+  split,
+  { by_cases h : s = ∅ ∨ t = ∅,
+    { simp [h] },
+    { simp [h],
```

Comment:
Rather than this non-terminal `simp`, how about doing the `assume` first, then use `left` or `apply or.inl` to change the goal appropriately. Same in the other proof.

Reply:
Right. Done.